### PR TITLE
Fix#110

### LINKED ISSUE #110: Manager Container restarts due to slow API
We are seeing an issue where our manager containers restarts due to slow API response:

```
E0114 17:33:16.025717       1 leaderelection.go:325] error retrieving resource lock openshift-operators/b7831733.redhat.io: Get "https://redacted_ip:443/api/v1/namespaces/openshift-operators/configmaps/b7831733.redhat.io": context deadline exceeded

I0114 17:33:16.025982       1 leaderelection.go:278] failed to renew lease openshift-operators/b7831733.redhat.io: timed out waiting for the condition
```
This creates a downstream problem for us as cert-utils reconciles resources on startup and makes changes to the secrets which causes reloader (https://github.com/stakater/Reloader) to think the cert has changed, when it hasn't. This causes mass app pod restarts.

By default, the operator runs in a single instance so there really is no need for a leader-election process.

Would it be possible to do one of the following:

1. Allow the timeout to be configurable through a CR (Give API server more time to respond)
2. Disable leader-election through CR 
3. Is there a way to modify the behavior of cert-utils from changing secrets when it reconciles on startup?

### PR DESCRIPTION
**Pull request by raffaelespazzoli**
fixes #110

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/raffaelespazzoli/cert-utils-operator
```
Comment (sabre1041) [inline config/manager/kustomization.yaml:15]: shouldnt it be `quay.io/redhat-cop/cert-utils-operator

**Review by sabre1041 — CHANGES_REQUESTED** [review]

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (raffaelespazzoli) [reply config/manager/kustomization.yaml:15]: that field is all over the place in the other operators, probably it does not matter. anyway I fixed it

Comment (sabre1041) [reply config/manager/kustomization.yaml:15]: agree. only reason I called this one out was that it was not a valid image path. Thanks for updating.

**Review by sabre1041 — APPROVED** [review]
LGTM

**Comment by jdziedzic** [comment]
Looks like this may have inadvertently caused another issue. One of my developers is seeing an issue with the private key password. 

```
Looking at the source code of cert-utils-operator which we use to generate keystore from tls secrets – I noticed that it is changed recently to fix frequent restarts of the pods every time the operator is restarted.. Below is the issue and the commit that fixes it.
 
https://github.com/redhat-cop/cert-utils-operator/issues/110
https://github.com/redhat-cop/cert-utils-operator/commit/c55ca62a3d9b67529476744c07eaa9097a648486#diff-266e243251ed42b2e1bcf54f2a041cdd69176644a9f4b4e2ec140edd32ea56e3L179
 
But the problem here is when the private key is being created and added to the keystore – it is added without any password – this line: https://github.com/redhat-cop/cert-utils-operator/blob/c55ca62a3d9b67529476744c07eaa9097a648486/controllers/secrettokeystore/secret_to_keystore_controller.go#L265
 
It needs to be changed to []byte(getPassword(secret)) instead of []byte{} – which means it is setting up empty password for the private key. And to test my hypothesis – I tried accessing the private key inside the keystore using an empty password and I was able to access it just fine.
 
So to sum it up – Keystore password is set to “jbosscert” and key password is set to empty string when the JKS files are generated.
 
```
